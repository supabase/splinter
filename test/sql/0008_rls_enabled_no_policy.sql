begin;
  set local search_path = '';

  create table public.blog(
    id int primary key
  );

  alter table public.blog enable row level security;

  -- 1 issue
  select * from lint."0008_rls_enabled_no_policy";

  -- resolve the issue
  create policy none_shall_pass on public.blog
  for select
  using (false);

  -- 0 issue
  select * from lint."0008_rls_enabled_no_policy";

rollback;
