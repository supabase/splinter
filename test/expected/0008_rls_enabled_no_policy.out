begin;
  set local search_path = '';
  create table public.blog(
    id int primary key
  );
  alter table public.blog enable row level security;
  -- 1 issue
  select * from lint."0008_rls_enabled_no_policy";
         name          |         title         | level |  facing  | categories |                                                   description                                                   |                            detail                            |                                        remediation                                        |                       metadata                        |             cache_key             
-----------------------+-----------------------+-------+----------+------------+-----------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------+-------------------------------------------------------------------------------------------+-------------------------------------------------------+-----------------------------------
 rls_enabled_no_policy | RLS Enabled No Policy | INFO  | EXTERNAL | {SECURITY} | Detects cases where row level security (RLS) has been enabled on a table but no RLS policies have been created. | Table \`public.blog\` has RLS enabled, but no policies exist | https://supabase.com/docs/guides/database/database-linter?lint=0008_rls_enabled_no_policy | {"name": "blog", "type": "table", "schema": "public"} | rls_enabled_no_policy_public_blog
(1 row)

  -- resolve the issue
  create policy none_shall_pass on public.blog
  for select
  using (false);
  -- 0 issue
  select * from lint."0008_rls_enabled_no_policy";
 name | title | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
